lib/OpenQA/Worker/WebUIConnection.pm

```diff
@@ -194,23 +194,23 @@ sub _setup_websocket_connection {
                     $command_handler->handle_command(@_);
                 });
             $tx->on(
-                finish => sub {
+                finish => sub ($tx, $code, $reason = 'no reason') {
```

Comment:
You need `use Mojo::Base 'Mojo::EventEmitter', -signatures` to be able to use that syntax here.